Packs/CrowdStrikeFalcon/Integrations/CrowdStrikeFalcon/CrowdStrikeFalcon.py

```diff
@@ -2660,8 +2669,413 @@ def fetch_incidents():
+        seconds_to_add (int): The amount of seconds to add to the date.
+        date_format (str): The date format.
+
+    Returns:
+        str: The date with an increase in seconds.
+    """
+    added_datetime = datetime.strptime(date, date_format) + timedelta(seconds=seconds_to_add)
+    return added_datetime.strftime(date_format)
+
+
+def create_incident_from_ioa_event(ioa_event: dict[str, Any]) -> dict[str, Any]:
+    """Create an incident from an IOA event.
+
+    Args:
+        ioa_event (dict[str, Any]): An IOA event.
+
+    Returns:
+        dict[str, Any]: An incident from an IOA event.
+    """
+    ioa_event['mirror_direction'] = MIRROR_DIRECTION
+    ioa_event['mirror_instance'] = INTEGRATION_INSTANCE
+    ioa_event['extracted_account_id'] = demisto.get(ioa_event, 'cloud_account_id.aws_account_id') or \
+        demisto.get(ioa_event, 'cloud_account_id.azure_account_id')
+    resource = demisto.get(ioa_event, 'aggregate.resource', {})
+    id = resource.get('id', [])
+    uuid = resource.get('uuid', [])
+    ioa_event['extracted_uuid'] = uuid[0] if uuid else None
+    ioa_event['extracted_resource_id'] = id[0] if id else None
+    incident_context = {
+        'name': f'IOA Event ID: {ioa_event.get("event_id")}',
+        'rawJSON': json.dumps(ioa_event)
+    }
+    return incident_context
+
+
+def ioa_events_pagination(ioa_fetch_query: str, api_limit: int, ioa_next_token: str | None,
+                          fetch_limit: int = INCIDENTS_PER_FETCH) -> tuple[list[dict[str, Any]], str | None]:
+    """This is in charge of doing the pagination process in a single fetch run, since the fetch limit can be greater than
+    the api limit, in such a case, we do multiple API calls until we reach the fetch limit, or no more results are found by the API.
+
+    Args:
+        ioa_fetch_query (str): The IOA fetch query.
+        api_limit (int): The API limit
+        ioa_next_token (str | None): The IOA next token to start the pagination from.
+        fetch_limit (int, optional): The fetch limit. Defaults to INCIDENTS_PER_FETCH.
+
+    Returns:
+        tuple[list[dict[str, Any]], str | None]: A tuple where the first element is the fetched events, and the second is the next token that
+        will be used in the next fetch run.
+    """
+    total_incidents_count = 0
+    ioa_new_next_token = ioa_next_token
+    fetched_ioa_events: list[dict[str, Any]] = []
+    continue_pagination = True
+    while continue_pagination:
+        demisto.debug(f'Doing IOA pagination with the arguments: {ioa_fetch_query=}, {api_limit=}, {ioa_new_next_token=},'
+                      f'{fetch_limit=}')
+        ioa_events, ioa_new_next_token = get_ioa_events_for_fetch(ioa_fetch_query=ioa_fetch_query,
+                                                                  ioa_next_token=ioa_new_next_token,
+                                                                  limit=min(api_limit, fetch_limit - total_incidents_count))
```

Comment:
No, since we have two scenarios:
1. api_limit >= fetch_limit, in that case, limit to be fetch_limit (since total_incidents_count will be zero), and only do one pagination
2. fetch_limit > api_limit, in that case, limit will take the value api_limit until we reach a point where api_limit + counter > fetch_limit, and once we reach that case, limit will have the value (fetch_limit - counter), which is smaller than api_limit, meaning we will only do one API call in this situation, and then finish the pagination process.
If you feel that it's necessary, we can add a validation, where if this goes below 0, raise an Error. WDYT?

Reply:
That's OK, no need for a validation, thanks.